[recipes-core,initrdscripts] Persist aosupdate logs

mlohvynenko opened the pull request.

al1img commented on recipes-core/initrdscripts/initramfs-framework/aosupdate:
```diff
@@ -19,11 +19,28 @@ UPDATE_IMAGE=""
 UPDATE_DIR="${bootparam_aosupdate_disk:+${UPDATE_MOUNT}/}${bootparam_aosupdate_path:-update_rootfs}"
 UPDATE_EXT="*.squashfs"
 
+log() {
```
Better to use existin function for log output. I believe there is another solution to redirect it to systems journal. 

See:

poky/meta/recipes-core/initrdscripts/initramfs-framework/init
meta-aos-vm/meta-aos-vm-common/recipes-bsp/grub/grub-config.bb

I guess `console` or some other kernel param should solve the issue.

mykola-kobets-epam commented on recipes-core/initrdscripts/initramfs-framework/aosupdate:
```diff
@@ -210,21 +229,21 @@ do_apply() {
-            if [ -n "${bootparam_aosupdate_disk}" ]; then
-                mount_update_disk
-            fi
+    if [ -n "${bootparam_aosupdate_disk}" ]; then
```
please fix alignment

al1img commented:
Overriding existing initramfs framework log methods is a better approach. To do it, you have to create script files with overridden log methods and add it to the initramfs. See `poky/meta/recipes-core/initrdscripts/initramfs-framework/init` `msg`, `info`. `debug` and `fatal` methods. Create `meta-aos/recipes-core/initrdscripts/initramfs-framework/kmsglog` file with following content:

```console
# This file contains overridden log methods to redirect initramfs messages to /dev/kmsg

# Prints information
msg() {
    echo "<5>initramfs: $*" >/dev/kmsg
}

# Prints information if verbose bootparam is used
info() {
    [ -n "$bootparam_verbose" ] && echo "<6>initramfs: $*" >/dev/kmsg
}

# Prints information if debug bootparam is used
debug() {
    [ -n "$bootparam_debug" ] && echo "<7>initramfs: $*" >/dev/kmsg
}

# Prints a message and start a endless loop
fatal() {
    echo "<2>initramfs: $1" >/dev/kmsg
    echo "<2>initramfs:" >/dev/kmsg

    if [ -n "$bootparam_init_fatal_sh" ]; then
        sh
    else
        while [ "true" ]; do
            sleep 3600
        done
    fi
}

```

Then make it as part of initramfs framework with highest priority, see: `meta-aos/recipes-core/initrdscripts/initramfs-framework_%.bbappend`. Add it to `AOS_INITRAMFS_SCRIPTS` in `meta-aos/recipes-core/images/aos-image-initramfs.bb`.

The benefit of this approach - no need to modify existing initramfs scripts. Alls messages will go to kmsg.

mlohvynenko committed:
[initramfs] Redirect logs to kmseg

This patch ensures that all logs from initramfs are redirected to kmseg,
so they are visible in the kernel log buffer and journalctl.

Signed-off-by: Mykhailo Lohvynenko <Mykhailo_Lohvynenko@epam.com>
